src/sim.rs

```diff
@@ -380,12 +400,12 @@ impl<'a> Sim<'a> {
         }
 
         self.elapsed += tick;
-        self.steps += 1;
+        self.steps.fetch_add(1, Relaxed);
 
         if self.elapsed > self.config.duration && !is_finished {
             return Err(format!(
                 "Ran for duration: {:?} steps: {} without completing",
-                self.config.duration, self.steps,
+                self.config.duration, self.steps.load(Relaxed),
```

Comment:
You can avoid this load because fetch_add above returns the previous value so you can just +1 that and get the current value.

Reply:
Nice, will adjust that!